src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -482,44 +482,49 @@ private String getPrincipalName(String username, String domainName) {
         if (userDN.contains("/")) {
             userDN = userDN.replace("/","\\/");
         }
-        LOGGER.finer("Looking up group of "+userDN);
+        Set<GrantedAuthority> groups = new HashSet<GrantedAuthority>();
+
+        LOGGER.log(Level.FINER, "Looking up group of {0}", userDN);
         Attributes id = context.getAttributes(userDN,new String[]{"tokenGroups","memberOf","CN"});
         Attribute tga = id.get("tokenGroups");
-        if (tga==null) {// see JENKINS-11644. still trying to figure out when this happens
-            LOGGER.warning("Failed to retrieve tokenGroups for "+userDN);
-            HashSet<GrantedAuthority> r = new HashSet<GrantedAuthority>();
-            r.add(new GrantedAuthorityImpl("unable-to-retrieve-tokenGroups"));
-            return r;
-        }
-
-        // build up the query to retrieve all the groups
-        StringBuilder query = new StringBuilder("(|");
-        List<byte[]> sids = new ArrayList<byte[]>();
 
-        NamingEnumeration<?> tokenGroups = tga.getAll();
-        while (tokenGroups.hasMore()) {
-            byte[] gsid = (byte[])tokenGroups.next();
-            query.append("(objectSid={"+sids.size()+"})");
-            sids.add(gsid);
+        if (tga==null) {
+            // tga will be null if you are not using a global catalogue
+            // or if the user is not actually a member of any security groups.
+            LOGGER.log(Level.FINE, "Failed to retrieve tokenGroups for {0}", userDN);
```

Comment:
I am not sure about this message. If you read it, it looks like something is wrong, but most of the times you will get it because there are not any Security Groups. Perhaps, we should indicate that it can also be that there are not any tokenGroups?